app/src/main/java/com/sanjeevyadavit/magecart/presentation/products/ProductListFragment.kt

```diff
@@ -0,0 +1,56 @@
+package com.sanjeevyadavit.magecart.presentation.products
+
+import android.os.Bundle
+import android.util.Log
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.compose.ui.platform.ComposeView
+import androidx.fragment.app.activityViewModels
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.navArgs
+import com.sanjeevyadavit.magecart.common.components.StateContainer
+import com.sanjeevyadavit.magecart.domain.use_case.GetProductListUseCase
+import com.sanjeevyadavit.magecart.ui.products.ProductList
+import com.sanjeevyadavit.magecart.presentation.MainActivityViewModel
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class ProductListFragment : Fragment() {
+
+    // FIXME: I dont think this is the right way
+    @Inject lateinit var productListUseCase: GetProductListUseCase
```

Comment:
Use case should be injected into your viewModel and then be used. You fragment should call teh viewModel function and then viewModel's function should call the useCase and get the job done.